app/stock/analysis.py

```diff
@@ -121,10 +123,12 @@ def analyze(holding):
             if available_quantity > 1:
                 # sell all shares to stop loss
                 # but keep one to monitor on mobile app
-                shares = max(0, int(holding['quantity'] - 1))
+                shares = max(0, available_quantity - 1)
                 suggestion = {
                     'trade_type': trade.TradeType.sell,
-                    'shares': shares}
+                    'shares': shares,
+                    'reasons': "price is going down, sell to stop loss"}
```

Comment:
can you log the `daily_high`, `latest_price`, `available_quantity` and `settings.ACTION_DIFF_PERCENTAGE` in the reason?